sensing/autoware_cuda_pointcloud_preprocessor/src/cuda_pointcloud_preprocessor/cuda_pointcloud_preprocessor.cu

```diff
@@ -482,21 +464,21 @@ std::unique_ptr<cuda_blackboard::CudaPointCloud2> CudaPointcloudPreprocessor::pr
     thrust::device, device_ring_outlier_mask, device_ring_outlier_mask + num_organized_points_,
     device_indices);
 
-  std::uint32_t num_output_points;
-  cudaMemcpyAsync(
-    &num_output_points, device_indices + num_organized_points_ - 1, sizeof(std::uint32_t),
-    cudaMemcpyDeviceToHost, stream_);
+  int num_output_points{};
+  CHECK_CUDA_ERROR(cudaMemcpyAsync(
+    &num_output_points, device_indices + num_organized_points_ - 1, sizeof(int),
+    cudaMemcpyDeviceToHost, stream_));
 
-  cudaStreamSynchronize(stream_);
+  CHECK_CUDA_ERROR(cudaStreamSynchronize(stream_));
 
   // Get information and extract points after filters
-  std::uint32_t num_crop_box_passed_points =
+  long num_crop_box_passed_points =
```

Comment:
The actual return type of `thrust::count` should depend on the iterator that is passed as an argument. Since `device_crop_mask_` is defined as `thrust::device_vector<std::uint32_t>`, and `long` is a signed value,  I feel using `std::uint32_t` is more intuitively proper. Do you have the intention to use `long`?

Reply:
The function returns a `typename thrust::iterator_traits<normal_iterator<device_ptr<unsigned int>>>::difference_type (aka long)` (there is no way this would ever be negative, so I wonder why it's a `difference_type` and not a `size_t`...). I'll change it to `ssize_t` as that expresses the usage as a size, without raising linter errors for signed/unsigned conversions.

Reply:
Addressed in 
5193d98